app/src/behaviors/behavior_smart_interrupt.c

```diff
@@ -0,0 +1,288 @@
+/*
+ * Copyright (c) 2022 The ZMK Contributors
+ *
+ * SPDX-License-Identifier: MIT
+ */
+
+#define DT_DRV_COMPAT zmk_behavior_smart_interrupt
+
+#include <device.h>
+#include <drivers/behavior.h>
+#include <logging/log.h>
+#include <zmk/behavior.h>
+#include <zmk/keymap.h>
+#include <zmk/matrix.h>
+#include <zmk/event_manager.h>
+#include <zmk/events/position_state_changed.h>
+#include <zmk/events/layer_state_changed.h>
+#include <zmk/hid.h>
+
+LOG_MODULE_DECLARE(zmk, CONFIG_ZMK_LOG_LEVEL);
+
+#define ZMK_BHV_MAX_ACTIVE_SMART_INTERRUPTS 10
+
+struct behavior_smart_interrupt_config {
+    int32_t shared_key_positions_len;
+    int32_t shared_layers_len;
+    struct zmk_behavior_binding *behaviors;
```

Comment:
Since we require 3 behavior bindings exactly, I would break them out to explicit fields for each binding, e.g.:

```suggestion
    struct zmk_behavior_binding start_behavior;
    struct zmk_behavior_binding continue_behavior;
    struct zmk_behavior_binding end_behavior;
```

So you're not doing odd access by index work throughout the rest of the code.